client/components/editor.vue

```diff
@@ -598,42 +610,92 @@
+        let posMarks = mode === "after" ? posNode.marks : posNode.marks;
+        let hasHighlight = posMarks.length !== 0;
+        while (hasHighlight) {
+          for (let i = 0; i < posMarks.length; i += 1) {
+            const x = posMarks[i];
+            const highlightKeys = x.attrs["highlight-keys"].split(",");
+            let otherKeys = highlightKeys.filter((y) => {
+              return y !== commentHighlightKey;
+            });
+            otherKeys = otherKeys || [];
+            if (otherKeys.length < highlightKeys.length) {
+              if (mode === "after") {
+                highlightNodes.push({pos, otherKeys});
+                pos = this.$editorView.state.doc.resolve(pos.pos + posNode.nodeSize);
+                posNode = pos.nodeAfter;
+                posMarks = pos.nodeAfter.marks;
+              }
+              else {
+                highlightNodes.unshift({pos, otherKeys});
+                pos = this.$editorView.state.doc.resolve(pos.pos - pos.textOffset - 1);
+                posNode = pos.nodeBefore;
+                posMarks = pos.nodeBefore.marks;
+              }
+              hasHighlight = posMarks.length !== 0;
+            }
+            else {
+              hasHighlight = false;
+            }
+          }
         }
       },
 
-      filterComments(keys) {
-        if (!Meteor.userId()) {
-          return;
+      deleteCommentHighlight(comment, deleteHighlight) {
+        this.commentHKey = comment.highlightKey;
+        this.commentToDelete = comment;
+        if (deleteHighlight) {
+          const {doc, tr} = this.$editorView.state;
+          const cursorPos = this.$editorView.state.doc.resolve(this.$editorView.state.selection.$cursor.pos);
+          const highlightNodes = [];
+          this.getNearbyHighlights(highlightNodes, cursorPos, "after", comment.highlightKey);
+          this.getNearbyHighlights(highlightNodes, cursorPos, "before", comment.highlightKey);
+          // Update nearby highlights.
+          removeHighlight(
+            schema, tr, doc, highlightNodes[0].pos.pos - highlightNodes[0].pos.textOffset,
+            highlightNodes[highlightNodes.length - 1].pos.pos + highlightNodes[highlightNodes.length - 1].pos.nodeAfter.nodeSize,
+            this.$editorView.dispatch,
+            );
+          highlightNodes.forEach((d, i) => {
+            if (d.otherKeys.length > 0) {
+              addHighlight(
+                d.otherKeys.join(','), schema, tr, d.pos.pos - d.pos.textOffset,
+                d.pos.pos + d.pos.nodeAfter.nodeSize, this.$editorView.dispatch,
+                );
+            }
+          });
+          this.$editorView.dispatch(tr);
         }
-        if (!this.$editorView.state) {
-          return;
+        else {
+          this.$emit("highlight-deleted", {id: this.commentToDelete._id, version: collab.getVersion(this.$editorView.state)});
+          this.commentHKey = undefined;
+          this.commentToDelete = undefined;
```

Comment:
Any particular reason to use `undefined`? I prefer using `null` for things which code logic sets, in comparison with `undefined` which is something that it was not set, but happens to be.